File: Php/app/Homegame/Details/HomegameDetailsModel.php
namespace app\Homegame\Details{

	use core\HomegamePageModel;
	use entities\Cashgame;
	use entities\Homegame;
	use Domain\Classes\User;
	use app\Urls\HomegameEditUrlModel;

	class HomegameDetailsModel extends HomegamePageModel {

		public $displayName;
		public $description;
		public $houseRules;
		public $editUrl;
		public $showEditLink;

		public function __construct(User $user,
									Homegame $homegame,
									$isInManagerMode,
									Cashgame $runningGame = null){
			parent::__construct($user, $homegame, $runningGame);
			displayName = $homegame.getDisplayName();
			description = $homegame.getDescription();
			houseRules = formatHouseRules($homegame.getHouseRules());
			editUrl = new HomegameEditUrlModel($homegame);
			showEditLink = $isInManagerMode;
		}

		private function formatHouseRules($houseRules){
			return nl2br($houseRules, false);
		}

	}

}

File: Web/Models/HomegameModels/Details/HomegameDetailsModel.cs
using Core.Classes;
using Web.Models.Url;

namespace Web.Models.HomegameModels.Details{

	public class HomegameDetailsModel : HomegamePageModel {

	    public string DisplayName { get; set; }
	    public string Description { get; set; }
	    public string HouseRules { get; set; }
	    public bool ShowHouseRules { get; set; }
	    public UrlModel EditUrl { get; set; }
	    public bool ShowEditLink { get; set; }

	    public HomegameDetailsModel(User user, Homegame homegame, bool isInManagerMode, Cashgame runningGame = null) : base(user, homegame, runningGame)
	    {
	        DisplayName = homegame.DisplayName;
			Description = homegame.Description;
			HouseRules = FormatHouseRules(homegame.HouseRules);
	        ShowHouseRules = !string.IsNullOrEmpty(HouseRules);
			EditUrl = new HomegameEditUrlModel(homegame);
			ShowEditLink = isInManagerMode;
	    }

		private string FormatHouseRules(string houseRules)
		{
		    if (houseRules == null)
		        return null;
		    return houseRules.Replace("\n\r", "<br />\n\r");
		}

        public override string BrowserTitle
        {
            get
            {
                return "Homegame Details";
            }
        }

	}

}